docs/updater.md

```diff
@@ -0,0 +1,102 @@
+* `gvk::swapchain_resized_event`: This event occurs when the swapchain's dimensions have changed.
+* `gvk::swapchain_changed_event`: This event occurs when the swapchain is recreated for any reason.
+* `gvk::swapchain_format_changed_event`: This event occurs when the image format of the swapchain has changed. Currently, this can be triggered when invoking `window::request_srgb_framebuffer` for the given window.
+* `gvk::swapchain_additional_attachments_changed_event`: This event occurs when window's additional attachments change. The set of additional attachments may change when invoking `window::set_additional_back_buffer_attachments`.
+* `gvk::concurrent_frames_count_changed_event`: This event occurs when the number of concurrent frames has been changed by the user. Currently, this the number of concurrent frames is modified by invoking `window::set_number_of_concurrent_frames`.
+* `gvk::destroying_graphics_pipeline_event`: This event occurs when an outdated graphics pipeline is destroyed.
+* `gvk::destroying_compute_pipeline_event`: This event occurs when an outdated compute pipeline is destroyed.
+* `gvk::destroying_ray_tracing_pipeline_event`: This event occurs when an outdated ray tracing pipeline is destroyed.
+* `gvk::destroying_image_event`: This event occurs when an outdated image is destroyed.
+* `gvk::destroying_image_view_event`: This event occurs when an outdated image view is destroyed.
+* `gvk::files_changed_event`: This event occurs when any of the given files have been modified on the file system.
+
+
+## How to use
+
+The event handling and the automatic updater mechanisms are available via `gvk::updater` class. The access to an updater is restricted to objects of type `gvk::invokee`. This means that each `gvk::invokee` object possesses its own `gvk::updater` instance, functioning in isolation from other updater instances. This updater can be made available by the following call:
+```
+mUpdater.emplace();
+```
+The general schema for using an updater is the following:
+
+```
+mUpdater.on(event... events).[update(...)|invoke(...)][.then_on(event... events).[update(...)|invoke(...)]]*
+```
+
+* `events`: a comma seperated list of event objects. The events are in a logical _or_ relationship, which means that the bound callbacks are invoked if _any_ of the events occur.
+* `on(event... events)`: bind the following up `update` or `invoke` call to a list of events.
+* `update(updatee... updatees)`: automatically updates the given comma seperated list of objects once the bound event occurs. The following resources can be updated:
+  * `avk::graphics_pipeline`
+  * `avk::compute_pipeline`
+  * `avk::ray_tracing_pipeline`
+  * `avk::image`
+  * `avk::image_view`
+* `invoke(handler... handlers)`: invokes the callbacks specified by the comma separated list once the bound event occurs. The following function signatures are allowed:
```

Comment:
```suggestion
* `invoke(handler... handlers)`: Comma-separated list of handlers which are invoked after _any_ of the previously defined events has occurred. The following function signatures are supported:
```

Reply:
Github turns Comma-separated to Comma separated. Cannot figure out why!